client/cluster_error.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+type ClusterError struct {
+	Errors []error
+}
+
+func (ce ClusterError) Error() string {
+	return ErrClousterNotAvailable.Error()
```

Comment:
yep, the net error has ip attached.

The context error doesn't have the endpoint attached so far. If client does one http request and deadline exceeded, he may have no idea which endpoint may be that slow.